tests/app/shared/issue-tables/search-filter.spec.ts

```diff
@@ -0,0 +1,89 @@
+import { Issue } from '../../../../src/app/core/models/issue.model';
+import { Team } from '../../../../src/app/core/models/team.model';
+import { DUPLICATED_ISSUE_WITH_EMPTY_DESCRIPTION_HIGH_SEVERITY, ISSUE_WITH_ASSIGNEES, ISSUE_WITH_EMPTY_DESCRIPTION,
+    ISSUE_WITH_EMPTY_DESCRIPTION_HIGH_SEVERITY, ISSUE_WITH_EMPTY_DESCRIPTION_LOW_SEVERITY } from '../../../constants/githubissue.constants';
+import { TABLE_COLUMNS } from '../../../../src/app/shared/issue-tables/issue-tables.component';
+import { IssueService } from '../../../../src/app/core/services/issue.service';
+import { applySearchFilter } from '../../../../src/app/shared/issue-tables/search-filter';
+
+describe('search-filter', () => {
+    describe('applySearchFilter()', () => {
+        const dummyTeam: Team = new Team({
+        id: 'dummyId',
+        teamMembers: [],
+        });
+        let dummySearchKey: string;
+        const mediumSeverityIssueWithResponse: Issue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION, dummyTeam);
+        mediumSeverityIssueWithResponse.responseTag = 'Accepted';
+        const mediumSeverityIssueWithAssigneee: Issue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_ASSIGNEES, dummyTeam);
+        const lowSeverityFeatureFlawIssue: Issue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION_LOW_SEVERITY, dummyTeam);
+        const highSeverityDocumentationBugIssue: Issue =
+            Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION_HIGH_SEVERITY, dummyTeam);
+        const duplicatedIssue: Issue = Issue.createPhaseTeamResponseIssue(DUPLICATED_ISSUE_WITH_EMPTY_DESCRIPTION_HIGH_SEVERITY, dummyTeam);
+        duplicatedIssue.duplicateOf = highSeverityDocumentationBugIssue.id;
+
+        const issuesList: Issue[] = [
+            mediumSeverityIssueWithResponse,
+            mediumSeverityIssueWithAssigneee,
+            lowSeverityFeatureFlawIssue,
+            highSeverityDocumentationBugIssue
+        ];
+        const displayedColumn: string[] = [
+            TABLE_COLUMNS.ID,
+            TABLE_COLUMNS.TITLE,
+            TABLE_COLUMNS.TYPE,
+            TABLE_COLUMNS.SEVERITY,
+            TABLE_COLUMNS.RESPONSE,
+            TABLE_COLUMNS.ASSIGNEE,
+            TABLE_COLUMNS.DUPLICATED_ISSUES,
+        ];
+        const issueService: IssueService = new IssueService(null, null, null, null, null, null);
+
+        beforeEach(() => {
+            issueService.updateLocalStore(mediumSeverityIssueWithResponse);
+            issueService.updateLocalStore(mediumSeverityIssueWithAssigneee);
+            issueService.updateLocalStore(lowSeverityFeatureFlawIssue);
+            issueService.updateLocalStore(highSeverityDocumentationBugIssue);
+            issueService.updateLocalStore(duplicatedIssue);
+        });
+
+        it('returns filtered list of issues which includes issues that contain the search key in any of its assignees', () => {
+            dummySearchKey = 'anubh-v';
```

Comment:
Could we improve upon this by creating `USER_ANUBHAV` in the `data.constants.ts` and then reference the `id` here and also in the `ISSUE_WITH_ASSIGNEES` and other issues with his `loginId`?

Reply:
Yup, sounds good!